runtime/module.go

```diff
@@ -251,6 +251,20 @@ func ProvideAppVersionModifier(app *AppBuilder) baseapp.AppVersionModifier {
 	return app.app
 }
 
+func ProvideEnvironment(config *runtimev1alpha1.Module, key depinject.ModuleKey, app *AppBuilder) appmodule.Environment {
+	env := appmodule.Environment{}
+
+	env.KvStoreService = ProvideKVStoreService(config, key, app)
```

Comment:
We should actually add KVStoreService as a function parameter to ProvideEnvironment and let depinject call ProvideKVStoreService. This ensures the store key is constructed only once in the dependency graph per module. Otherwise there's a risk of a hard to debug error if a module depends on both an environment and a store key.

Reply:
Yes, @aaronc has a good point here.

Reply:
like so https://github.com/cosmos/cosmos-sdk/pull/19041/commits/bbc3f3aca1ef2ae6aec02e94e5a50c6b16b9f321?

Reply:
No just that type as a parameter:
```
ProvideEnvironment(kvSvc KVStoreService)
```